### ISSUE TITLE
Rockstor does not handle drive/pool errors with Grace or Recognition

### ISSUE THREAD

**Issue by tastyratz** [rockstor/rockstor-core#1199]
I Did another break fix test.
vmware 5.5, unique UUID for vmdk's.

I shut down Rockstor and removed one of the drives in vmware, then booted it back up. When viewing the near empty raid5 pool I now see 89.37% usage and get an error:

```
Error! 
Error running a command. cmd = ['/bin/mount', u'/dev/disk/by-label/test2', u'/mnt2/test2', '-o', u',compress=no']. rc = 32. stdout = ['']. stderr = ['mount: wrong fs type, bad option, bad superblock on /dev/sdd,', ' missing codepage or helper program, or other error', '', ' In some cases useful info is found in syslog - try', ' dmesg | tail or so.', ''] 
If you need more help on this issue, email us at support@rockstor.com with the following information: 
• The error message: "Error running a command. cmd = ['/bin/mount', u'/dev/disk/by-label/test2', u'/mnt2/test2', '-o', u',compress=no']. rc = 32. stdout = ['']. stderr = ['mount: wrong fs type, bad option, bad superblock on /dev/sdd,', ' missing codepage or helper program, or other error', '', ' In some cases useful info is found in syslog - try', ' dmesg | tail or so.', '']"
• A step by step description of the actions leading up to the error.
• Download the tarred and zipped server logs by clicking here, and attach it to the email.

```

I can’t click into the degraded pool at all without that error happening so I have no way to view/repair/investigate/research. I see no degraded indicator, red highlight, or other reason looking at pools or share lists to think I might be having issues.

The removed drive even still shows up in the Disks list. I can click error logs/self-test logs and it shows “there are no errors”! Shouldn't it at least say the drive is missing or present some kind of error?

I see no other external indicator to tell me that a drive dropped out/failed. That kind of problem should be jumping out at me on the dashboard. I should probably get a pop up requiring acknowledgement before continuing. Console output is loaded with errors, but nothing in Rockstor webUI.

Rockstor should probably be actively polling drive status to report on these kinda of things.

**Comment by phillxnet** [comment]
@tastyratz An update on our coverage of the issues raised in this report:

"The removed drive even still shows up in the Disks list. I can click error logs/self-test logs and it shows “there are no errors”! Shouldn't it at least say the drive is missing or present some kind of error?"

As of all recent stable and testing channel releases detached disks are renamed with names beginning with "detached-..." and the SMART data as expressed in this issue is always historical (and we have an outstanding issue to improve messaging on this point: see #1435).

Linking to another related open issue re pool status reporting #1531.

**Comment by phillxnet** [comment]
"I see no other external indicator to tell me that a drive dropped out/failed." has now been addressed via the following pr:
"surface pool missing disk info in UI. Fixes #1897" #1902
